packages/components/src/custom-select-control/index.js

```diff
@@ -81,6 +101,18 @@ export default function CustomSelectControl( {
 		stateReducer,
 	} );
 
+	const anchorRef = useRef();
+
+	// Calculate Popover width based on the size of the component's container.
+	// The width will be set for the content of the menu, so factor in the
+	// Popover's border to ensure that the menu lines up with the toggle button.
+	const [
+		containerResizeListener,
+		{ width: containerWidth },
+	] = useResizeObserver();
+	const widthMinusBorder =
+		containerWidth >= 2 ? containerWidth - 2 : containerWidth;
```

Comment:
I wonder if there's a way to get the `border-width` of the `Popover` instead of hardcoding it. Maybe we could get a `ref` of the `Popover` and get the `computedStyle`s ?

Reply:
Or maybe we could use a different value of `box-sizing` so that we don't need to take the border width into account at all?

Reply:
This one was slightly tricky because the border is set on `.components-popover__content` which is a child of the `Popover` container, so a `ref` doesn't quite get us there. Also, because `.components-popover__content` is a _parent_ of the element we're attaching `menuProps` to, `box-sizing` doesn't help us either — the border is set on the parent so the width of the child cannot take it into account. Also, I wanted to avoid overriding Popover classes directly where we can.

I _think_ the neatest thing to do here is to actually skip the JS calculation that added in `2px` and instead set a negative horizontal margin that references `$border-width` so we're still matching a variable instead of using a magic number (I've seen this in a couple of other places in the repo, too). I've added that in this commit: https://github.com/WordPress/gutenberg/pull/37272/commits/27c0cc2bb5b9cf844474453f6c4144c2a571c088

Happy to continue to iterate there if you can think of a better way to handle it, but 🤞 this makes things a bit neater.